src/app/pages/cookies/cookies-modal/cookies-modal.component.ts

```diff
@@ -0,0 +1,55 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Inject, OnInit, Output } from '@angular/core';
+
+import { COOKIE_CONSENT_OPTIONS } from 'ish-core/configurations/injection-keys';
+import { CookieConsentOptions, CookieConsentSettings } from 'ish-core/models/cookies/cookies.model';
+import { CookiesService } from 'ish-core/services/cookies/cookies.service';
+
+/**
+ * Cookies Modal Component
+ */
+@Component({
+  selector: 'ish-cookies-modal',
+  templateUrl: './cookies-modal.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CookiesModalComponent implements OnInit {
+  @Output() close = new EventEmitter<void>();
+
+  cookieConsentSettings?: CookieConsentSettings;
+  selectedIds = {};
+
+  // tslint:disable:no-intelligence-in-artifacts
+  constructor(
+    @Inject(COOKIE_CONSENT_OPTIONS) public cookieConsentOptions: CookieConsentOptions,
+    private cookiesService: CookiesService
+  ) {}
+
+  ngOnInit() {
+    this.cookieConsentSettings = JSON.parse(this.cookiesService.get('cookieConsent') || 'null');
+    Object.keys(this.cookieConsentOptions.options).forEach(option => {
+      if (
+        this.cookieConsentOptions.options[option].required ||
+        this.cookieConsentSettings?.enabledOptions.includes(option)
+      ) {
+        this.selectedIds[option] = true;
+      }
+    });
+  }
+
+  acceptAll() {
+    this.cookiesService.setCookiesConsentForAll();
+  }
+
+  submit() {
+    this.cookiesService.setCookiesConsentFor(
+      Object.keys(this.selectedIds).reduce((acc, x) => (this.selectedIds[x] ? acc.push(x) && acc : acc), [])
+    );
+    this.close.emit();
+  }
+
+  hide() {
+    this.close.emit();
+  }
+
+  unsorted = () => -1;
```

Comment:
Just as reminder from the meeting: if this would work correctly, the sorting order would be reversed.

Reply:
Keeps the intended order in Firefox and reverses the order in Chrome. It would be to easy. Returning `1` instead of `-1` just reverses the order in Firefox and is correct in Chrome.
Returning `0` seems to be agreeable by both browser engines resulting in the original order.